### PR TITLE
Introduce transforms of obb/oriented beam and conversion from aabb to obb.

### PR DESCRIPTION
**Pull request by feuerste**
This is required to support passing local transforms to iterators.

### CODE REVIEW

File: src/math.rs [hunk 1]
```diff
@@ -225,17 +264,22 @@ impl OrientedBeam {
...
+        isometry: &Isometry3<f64>,
         half_extent: &Vector2<f64>,
     ) -> [Point3<f64>; 4] {
-        let transform =
-            |x: f64, y: f64| rotation.rotate_point(Point3::new(x, y, 0.0)) + translation;
+        let transform = |x: f64, y: f64| {
+            isometry.rotation.rotate_point(Point3::new(x, y, 0.0)) + isometry.translation
+        };
         [
             transform(half_extent.x, half_extent.y),
```
Comment (catevita) [inline src/math.rs]: could be this transform be confused with the function in line 272? could it be more explicit like get_corner_from_offset

Comment (feuerste) [reply src/math.rs]: Yes, definitely, I renamed it.

File: src/octree/batch_iterator.rs [hunk 2]
```diff
@@ -91,36 +111,58 @@ where
...
+                        disp: global_from_local.translation,
+                    }) * frustum,
+                ),
+                PointCulling::OrientedBeam(beam) => {
+                    PointCulling::OrientedBeam(beam.transform(global_from_local))
+                }
+            },
+            None => location.culling.clone(),
         };
+        let local_from_global = location.global_from_local.clone().map(|t| t.inverse());
```
Comment (catevita) [inline src/octree/batch_iterator.rs]: what does it happen if location.global_from_local is None?

Comment (catevita) [reply src/octree/batch_iterator.rs]: sorry i found it, it maps Option<T> to Option<U

Comment (feuerste) [reply src/octree/batch_iterator.rs]: `map` leaves the `None` untouched, see https://doc.rust-lang.org/std/option/enum.Option.html#method.map
The point stream internally handles it accordingly.

**Comment by nnmm** [comment]
:+1:

**Comment by feuerste** [comment]
@wally-the-cartographer merge

**Pull request merged**